spec/requests/sales_logs_controller_spec.rb

```diff
@@ -352,14 +352,14 @@
               let(:user) { create(:user, organisation:) }
               let(:bulk_upload) { create(:bulk_upload, :sales, user:) }
 
-              let!(:included_log) { create(:sales_log, :in_progress, bulk_upload:, owning_organisation: organisation) }
-              let!(:excluded_log) { create(:sales_log, :in_progress, owning_organisation: organisation) }
+              let!(:included_log) { create(:sales_log, :in_progress, purchid: "included_id", bulk_upload:, owning_organisation: organisation) }
+              let!(:excluded_log) { create(:sales_log, :in_progress, purchid: "excluded_id", owning_organisation: organisation) }
 
               it "returns logs only associated with the bulk upload" do
                 get "/sales-logs?bulk_upload_id[]=#{bulk_upload.id}"
 
-                expect(page).to have_content(included_log.id)
-                expect(page).not_to have_content(excluded_log.id)
+                expect(page).to have_content(included_log.purchid)
```

Comment:
Can you just confirm that `purchid` is really what you want here? 
My understanding is that the context (and what the test was doing before) are looking at the log.id and the `purchid` is like an optional question answer on a log and not used as a primary key or anything like that.

Reply:
Ah, this is just a flaky test. So checking for log id is correct, but often fails because id can be any integer, and ofter it's being found somewhere else where we don't expect it. The purpose of this test is to check that a specific log appears or does not appear in the list, so checking the purchid should be enough to confirm that